docs/lists.rst

```diff
@@ -0,0 +1,502 @@
+that this works print list *squares*.
+
+Let's do this once more using list comprehensions:
+
+.. code-block:: python
+
+    squares2 = [i**2 for i in range(1,11)]
+
+
+If you print this, you get the exact same list, but we only needed one line of
+code instead of three. Let's now look at a slightly more complex example. We'll
+create a list of remainders when you divide the first 10 squares by 5.
+
+    To find the remainder when you divide by 5 use the ``%`` operator.
+
+.. code-block:: python
+
+    remainders = [(x ** 2) % 5 for x in range(1,11)]
+    print(remainders)
+
+    # this is the output
+    [1, 4, 4, 1, 0, 1, 4, 4, 1, 0]
+
+If you print the list, you'll see that there are only three perfect squares mod
+5: 0, 1 and 4. This example shows you that the expressions in the list
+comprehensions can be complex. By the way, if you look at the remainders when
+you divide by a prime number *p* you'll notice an interesting pattern: the
+number of remainders is (p+1)/2. The problem of finding which number appear in
+the list is a comple puzzle from number theory known as *quadratic reciprocity*
+and was first proved by Gauss.
+
+Next, let's create a list comprehension that has an if clause. Suppose we have
+a list of movies and we want to find those movies that start with the letter G.
+Let's see how to do this with and without lists comprehensions.
+
+If you're not using list comprehensions you'd start by making an empty list, next
+loop over the list of movies. We can use the ``startswith()`` method to see if
+the title starts with the letter G. If it does, then append it to out list.
+
+.. code-block:: python
+
+    movies = [
+        "Star Wars", "Ghandi", "Casablanca", "Shawshank Redemption",
+        "Toy Story", "Gone with the wind", "Citizen Kane", "It's a wonderful life",
+        "The Wizard of Oz", "Gattaca", "Rear Window", "Ghostbusters",
+        "To Kill a Mockingbird", "Good Will Hunting", "2001: A Space Odissey",
+        "Riders of the Lost Ark", "Groundhog Day",
+        "Close Encounters of the Third King", "Scent of a Woman",
+    ]
+
+    g_movies = []
+    for title in movies:
+        if title.startswith("G"):
+            g_movies.append(title)
+
+
+Print the list to make sure that it worked. But this four line routine can be
+done in a single line with a list comprehension. The expression we want to
+appear in our list is simply the title, next loop over the movies, but also
+check that the title starts with the letter G.
+
+.. code-block:: python
+
+    g_movies = [title for title in movies if title.startswith("G")]
+
+
+Print and observe: we get the same answer with a single line of code.
+
+    ``["Ghandi", "Gone with the wind", "Gattaca", "Ghostbusters",
+    "Good Will Hunting", "Groundhog Day"]``
+
+Let's complicate this example a bit more. Suppose our list of movies is a list
+of tuples containing both the title of the movie and the year it was released.
```

Comment:
again new concept and I think is too complicated for the beginning

Reply:
I would rather present the tuples before the lists. In many documentations lists, sets and tuples are presented altogether.
For the moment, I can replace the word tuple with pair and keep the reader focused on lists.